include/mbedtls/ssl_internal.h

```diff
@@ -143,32 +143,66 @@
 #define MBEDTLS_SSL_PADDING_ADD              0
 #endif
 
-#define MBEDTLS_SSL_PAYLOAD_LEN ( MBEDTLS_SSL_MAX_CONTENT_LEN    \
-                        + MBEDTLS_SSL_COMPRESSION_ADD            \
-                        + MBEDTLS_MAX_IV_LENGTH                  \
-                        + MBEDTLS_SSL_MAC_ADD                    \
-                        + MBEDTLS_SSL_PADDING_ADD                \
-                        )
+#define MBEDTLS_SSL_IN_PAYLOAD_LEN ( MBEDTLS_SSL_IN_CONTENT_LEN    \
+                        + MBEDTLS_SSL_COMPRESSION_ADD              \
+                        + MBEDTLS_MAX_IV_LENGTH                    \
+                        + MBEDTLS_SSL_MAC_ADD                      \
+                        + MBEDTLS_SSL_PADDING_ADD )                \
+
+#define MBEDTLS_SSL_OUT_PAYLOAD_LEN ( MBEDTLS_SSL_OUT_CONTENT_LEN    \
+                        + MBEDTLS_SSL_COMPRESSION_ADD                \
+                        + MBEDTLS_MAX_IV_LENGTH                      \
+                        + MBEDTLS_SSL_MAC_ADD                        \
+                        + MBEDTLS_SSL_PADDING_ADD )                  \
 
 /*
  * Check that we obey the standard's message size bounds
  */
 
-#if MBEDTLS_SSL_MAX_CONTENT_LEN > 16384
-#error Bad configuration - record content too large.
+#if MBEDTLS_SSL_IN_CONTENT_LEN > 16384
```

Comment:
Why don't we check `#if MBEDTLS_SSL_IN_CONTENT_LEN > MBEDTLS_SSL_MAX_CONTENT_LEN`?

We've defined a constant and are not using it.